rstudio/rstudio
# Fix regression in Visual Editor handling of raw html blocks

### Intent

Addresses #15189

### Approach

Use custom pandoc writer script to preserve the raw HTML blocks. See https://github.com/quarto-dev/quarto/pull/553 for essentially the same change in the VSCode Quarto extension.

### Automated Tests

None - I will add some tests in `main` when I forward-port this change.

### QA Notes

The modified code is invoked whenever the visual editor saves an R Markdown or Quarto file and when you switch between the Visual Editor and source mode.
 
Test RMD and QMD files containing raw HTML blocks and inline HTML, and switch back and forth from source mode to visual mode. Ensure the structure is preserved, especially that the raw block delimiters don't vanish, and inline HTML doesn't get modified. There may be whitespace changes when switching modes, that isn't new. Here's an example doc:

````qmd
---
title: Raw HTML
---

## Raw Block

```{=html}
Hello <i>World</i>, how are you?
```

## Inline HTML

Some more text and some <b>Bold</b> stuff.

````

### Documentation

NA

### Checklist
